// dapr-client/actors/runtime/ActorTimerData.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var BufferSerializer_1 = __importDefault(require("./BufferSerializer"));
/**
 * Contains the actor reminder data
 */
var ActorTimerData = /** @class */ (function () {
    /**
     * @param reminderName the name of the actor reminder
     * @param state the state data passed to receiveReminder callback
     * @param callback the callback method to call
     */
    function ActorTimerData(timerName, callback, state) {
        this.timerName = timerName;
        this.callback = callback;
        this.state = state;
    }
    ActorTimerData.prototype.getTimerName = function () {
        return this.timerName;
    };
    ActorTimerData.prototype.getState = function () {
        return this.state;
    };
    ActorTimerData.prototype.getCallback = function () {
        return this.callback;
    };
    /**
     * Return this class as an object
     */
    ActorTimerData.prototype.toObject = function () {
        return {
            callback: this.callback,
            data: this.state
        };
    };
    ActorTimerData.fromObject = function (reminderName, obj) {
        var serializer = new BufferSerializer_1.default();
        var callback = obj === null || obj === void 0 ? void 0 : obj.callback;
        var data = obj === null || obj === void 0 ? void 0 : obj.data;
        var deserializedData = serializer.deserialize(data);
        return new ActorTimerData(reminderName, callback, deserializedData);
    };
    return ActorTimerData;
}());
exports.default = ActorTimerData;
