// dapr-client/actors/runtime/ActorReminderData.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var BufferSerializer_1 = __importDefault(require("./BufferSerializer"));
/**
 * Contains the actor reminder data
 */
var ActorReminderData = /** @class */ (function () {
    /**
     * @param reminderName the name of the actor reminder
     * @param state the state data passed to receiveReminder callback
     * @param dueTime the amount of time to delay before invoking the reminder for the first time
     * @param period the time interval between reminder invocations after the first invocation
     */
    function ActorReminderData(reminderName, dueTime, period, state) {
        this.reminderName = reminderName;
        this.dueTime = dueTime;
        this.period = period;
        this.state = state;
    }
    ActorReminderData.prototype.getReminderName = function () {
        return this.reminderName;
    };
    ActorReminderData.prototype.getState = function () {
        return this.state;
    };
    ActorReminderData.prototype.getDueTime = function () {
        return this.dueTime;
    };
    ActorReminderData.prototype.getPeriod = function () {
        return this.period;
    };
    /**
     * Return this class as an object
     */
    ActorReminderData.prototype.toObject = function () {
        return {
            reminderName: this.reminderName,
            dueTime: this.dueTime,
            period: this.period,
            data: this.state
        };
    };
    ActorReminderData.fromObject = function (reminderName, obj) {
        var serializer = new BufferSerializer_1.default();
        var data = obj === null || obj === void 0 ? void 0 : obj.data;
        var dueTime = obj === null || obj === void 0 ? void 0 : obj.dueTime;
        var period = obj === null || obj === void 0 ? void 0 : obj.period;
        var deserializedData = serializer.deserialize(data);
        return new ActorReminderData(reminderName, dueTime, period, deserializedData);
    };
    return ActorReminderData;
}());
exports.default = ActorReminderData;
