// dapr-client/implementation/Server/GRPCServer/GRPCServer.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var grpc = __importStar(require("@grpc/grpc-js"));
var GRPCServerImpl_1 = __importDefault(require("./GRPCServerImpl"));
var appcallback_grpc_pb_1 = require("../../../proto/dapr/proto/runtime/v1/appcallback_grpc_pb");
var GRPCServer = /** @class */ (function () {
    function GRPCServer() {
        this.isInitialized = false;
        this.serverHost = "";
        this.serverPort = "";
        // Create Server
        this.server = new grpc.Server();
        this.serverCredentials = grpc.ServerCredentials.createInsecure();
        this.serverImpl = new GRPCServerImpl_1.default();
        // Add our implementation
        console.log("[Dapr-JS][gRPC] Adding Service Implementation - AppCallbackService");
        // @ts-ignore
        this.server.addService(appcallback_grpc_pb_1.AppCallbackService, this.serverImpl);
    }
    GRPCServer.prototype.getServerAddress = function () {
        if (!this.isInitialized) {
            throw new Error(JSON.stringify({
                error: "GRPC_SERVER_NOT_INITIALIZED",
                error_message: "The gRPC server was not initialized, did you call `await GRPCServerSingleton.initialize()`?"
            }));
        }
        return this.serverHost + ":" + this.serverPort;
    };
    GRPCServer.prototype.getServer = function () {
        if (!this.isInitialized) {
            throw new Error(JSON.stringify({
                error: "GRPC_SERVER_NOT_INITIALIZED",
                error_message: "The gRPC server was not initialized, did you call `await GRPCServerSingleton.initialize()`?"
            }));
        }
        return this.server;
    };
    // We allow this, since this will register the routes and handlers!
    GRPCServer.prototype.getServerImpl = function () {
        return this.serverImpl;
    };
    GRPCServer.prototype.close = function () {
        return __awaiter(this, void 0, void 0, function () {
            var _this = this;
            return __generator(this, function (_a) {
                if (!this.isInitialized) {
                    throw new Error(JSON.stringify({
                        error: "GRPC_SERVER_NOT_INITIALIZED",
                        error_message: "The gRPC server was not initialized, did you call `await GRPCServerSingleton.initialize()`?"
                    }));
                }
                return [2 /*return*/, new Promise(function (resolve, reject) {
                        _this.server.tryShutdown(function (err) {
                            if (err) {
                                return reject(err);
                            }
                            console.log("[Dapr-JS][gRPC] Closed Server");
                            _this.isInitialized = false;
                            return resolve();
                        });
                    })];
            });
        });
    };
    GRPCServer.prototype.startServer = function (host, port) {
        return __awaiter(this, void 0, void 0, function () {
            var delayMs;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        this.serverHost = host;
                        this.serverPort = port;
                        return [4 /*yield*/, this.initializeBind()];
                    case 1:
                        _a.sent();
                        this.server.start();
                        // We need to call the Singleton to start listening on the port, else Dapr will not pick it up correctly
                        // Dapr will probe every 50ms to see if we are listening on our port: https://github.com/dapr/dapr/blob/a43712c97ead550ca2f733e9f7e7769ecb195d8b/pkg/runtime/runtime.go#L1694
                        console.log("[Dapr-JS][gRPC] Letting Dapr pick-up the server");
                        delayMs = 250;
                        return [4 /*yield*/, (new Promise(function (resolve) { return setTimeout(resolve, delayMs); }))];
                    case 2:
                        _a.sent();
                        // We are initialized
                        this.isInitialized = true;
                        return [2 /*return*/];
                }
            });
        });
    };
    GRPCServer.prototype.initializeBind = function () {
        return __awaiter(this, void 0, void 0, function () {
            var _this = this;
            return __generator(this, function (_a) {
                console.log("[Dapr-JS][gRPC] Starting to listen on " + this.serverHost + ":" + this.serverPort);
                return [2 /*return*/, new Promise(function (resolve, reject) {
                        _this.server.bindAsync(_this.serverHost + ":" + _this.serverPort, _this.serverCredentials, function (err, port) {
                            if (err) {
                                return reject(err);
                            }
                            console.log("[Dapr-JS][gRPC] Listening on " + _this.serverHost + ":" + _this.serverPort);
                            return resolve();
                        });
                    })];
            });
        });
    };
    return GRPCServer;
}());
exports.default = GRPCServer;
