// dapr-client/actors/client/ActorProxy.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var ActorId_1 = __importDefault(require("../ActorId"));
var ActorProxyFactory_1 = __importDefault(require("./ActorProxyFactory"));
var ActorProxy = /** @class */ (function () {
    function ActorProxy(daprClient, actorType, actorId, actorInterface) {
        this.daprClient = daprClient;
        this.actorType = actorType;
        this.actorId = actorId;
        this.actorInterface = actorInterface;
        this.actorProxyFactory = new ActorProxyFactory_1.default(daprClient.getDaprHost(), daprClient.getDaprPort());
    }
    ActorProxy.prototype.getActorId = function () {
        return this.actorId;
    };
    ActorProxy.prototype.getActorType = function () {
        return this.actorType;
    };
    ActorProxy.create = function (client, actorType, actorId, actorInterface) {
        var factory = new ActorProxyFactory_1.default(client.getDaprHost(), client.getDaprPort());
        return factory.create(actorType, new ActorId_1.default(actorId), actorInterface);
    };
    return ActorProxy;
}());
exports.default = ActorProxy;
