// dapr-client/proto/dapr/proto/runtime/v1/appcallback_pb.js
// source: dapr/proto/runtime/v1/appcallback.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global = Function('return this')();

var google_protobuf_empty_pb = require('google-protobuf/google/protobuf/empty_pb.js');
goog.object.extend(proto, google_protobuf_empty_pb);
var dapr_proto_common_v1_common_pb = require('../../../../dapr/proto/common/v1/common_pb.js');
goog.object.extend(proto, dapr_proto_common_v1_common_pb);
goog.exportSymbol('proto.dapr.proto.runtime.v1.BindingEventRequest', null, global);
goog.exportSymbol('proto.dapr.proto.runtime.v1.BindingEventResponse', null, global);
goog.exportSymbol('proto.dapr.proto.runtime.v1.BindingEventResponse.BindingEventConcurrency', null, global);
goog.exportSymbol('proto.dapr.proto.runtime.v1.ListInputBindingsResponse', null, global);
goog.exportSymbol('proto.dapr.proto.runtime.v1.ListTopicSubscriptionsResponse', null, global);
goog.exportSymbol('proto.dapr.proto.runtime.v1.TopicEventRequest', null, global);
goog.exportSymbol('proto.dapr.proto.runtime.v1.TopicEventResponse', null, global);
goog.exportSymbol('proto.dapr.proto.runtime.v1.TopicEventResponse.TopicEventResponseStatus', null, global);
goog.exportSymbol('proto.dapr.proto.runtime.v1.TopicRoutes', null, global);
goog.exportSymbol('proto.dapr.proto.runtime.v1.TopicRule', null, global);
goog.exportSymbol('proto.dapr.proto.runtime.v1.TopicSubscription', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.dapr.proto.runtime.v1.TopicEventRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.dapr.proto.runtime.v1.TopicEventRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.dapr.proto.runtime.v1.TopicEventRequest.displayName = 'proto.dapr.proto.runtime.v1.TopicEventRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.dapr.proto.runtime.v1.TopicEventResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.dapr.proto.runtime.v1.TopicEventResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.dapr.proto.runtime.v1.TopicEventResponse.displayName = 'proto.dapr.proto.runtime.v1.TopicEventResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.dapr.proto.runtime.v1.BindingEventRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.dapr.proto.runtime.v1.BindingEventRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.dapr.proto.runtime.v1.BindingEventRequest.displayName = 'proto.dapr.proto.runtime.v1.BindingEventRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.dapr.proto.runtime.v1.BindingEventResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.dapr.proto.runtime.v1.BindingEventResponse.repeatedFields_, null);
};
goog.inherits(proto.dapr.proto.runtime.v1.BindingEventResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.dapr.proto.runtime.v1.BindingEventResponse.displayName = 'proto.dapr.proto.runtime.v1.BindingEventResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.dapr.proto.runtime.v1.ListTopicSubscriptionsResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.dapr.proto.runtime.v1.ListTopicSubscriptionsResponse.repeatedFields_, null);
};
goog.inherits(proto.dapr.proto.runtime.v1.ListTopicSubscriptionsResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.dapr.proto.runtime.v1.ListTopicSubscriptionsResponse.displayName = 'proto.dapr.proto.runtime.v1.ListTopicSubscriptionsResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.dapr.proto.runtime.v1.TopicSubscription = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.dapr.proto.runtime.v1.TopicSubscription, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.dapr.proto.runtime.v1.TopicSubscription.displayName = 'proto.dapr.proto.runtime.v1.TopicSubscription';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.dapr.proto.runtime.v1.TopicRoutes = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.dapr.proto.runtime.v1.TopicRoutes.repeatedFields_, null);
};
goog.inherits(proto.dapr.proto.runtime.v1.TopicRoutes, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.dapr.proto.runtime.v1.TopicRoutes.displayName = 'proto.dapr.proto.runtime.v1.TopicRoutes';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.dapr.proto.runtime.v1.TopicRule = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.dapr.proto.runtime.v1.TopicRule, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.dapr.proto.runtime.v1.TopicRule.displayName = 'proto.dapr.proto.runtime.v1.TopicRule';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.dapr.proto.runtime.v1.ListInputBindingsResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.dapr.proto.runtime.v1.ListInputBindingsResponse.repeatedFields_, null);
};
goog.inherits(proto.dapr.proto.runtime.v1.ListInputBindingsResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.dapr.proto.runtime.v1.ListInputBindingsResponse.displayName = 'proto.dapr.proto.runtime.v1.ListInputBindingsResponse';
}



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.dapr.proto.runtime.v1.TopicEventRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.dapr.proto.runtime.v1.TopicEventRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.dapr.proto.runtime.v1.TopicEventRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.dapr.proto.runtime.v1.TopicEventRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    id: jspb.Message.getFieldWithDefault(msg, 1, ""),
    source: jspb.Message.getFieldWithDefault(msg, 2, ""),
    type: jspb.Message.getFieldWithDefault(msg, 3, ""),
    specVersion: jspb.Message.getFieldWithDefault(msg, 4, ""),
    dataContentType: jspb.Message.getFieldWithDefault(msg, 5, ""),
    data: msg.getData_asB64(),
    topic: jspb.Message.getFieldWithDefault(msg, 6, ""),
    pubsubName: jspb.Message.getFieldWithDefault(msg, 8, ""),
    path: jspb.Message.getFieldWithDefault(msg, 9, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.dapr.proto.runtime.v1.TopicEventRequest}
 */
proto.dapr.proto.runtime.v1.TopicEventRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.dapr.proto.runtime.v1.TopicEventRequest;
  return proto.dapr.proto.runtime.v1.TopicEventRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.dapr.proto.runtime.v1.TopicEventRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.dapr.proto.runtime.v1.TopicEventRequest}
 */
proto.dapr.proto.runtime.v1.TopicEventRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setId(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setSource(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setType(value);
      break;
    case 4:
      var value = /** @type {string} */ (reader.readString());
      msg.setSpecVersion(value);
      break;
    case 5:
      var value = /** @type {string} */ (reader.readString());
      msg.setDataContentType(value);
      break;
    case 7:
      var value = /** @type {!Uint8Array} */ (reader.readBytes());
      msg.setData(value);
      break;
    case 6:
      var value = /** @type {string} */ (reader.readString());
      msg.setTopic(value);
      break;
    case 8:
      var value = /** @type {string} */ (reader.readString());
      msg.setPubsubName(value);
      break;
    case 9:
      var value = /** @type {string} */ (reader.readString());
      msg.setPath(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.dapr.proto.runtime.v1.TopicEventRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.dapr.proto.runtime.v1.TopicEventRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.dapr.proto.runtime.v1.TopicEventRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.dapr.proto.runtime.v1.TopicEventRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getSource();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getType();
  if (f.length > 0) {
    writer.writeString(
      3,
      f
    );
  }
  f = message.getSpecVersion();
  if (f.length > 0) {
    writer.writeString(
      4,
      f
    );
  }
  f = message.getDataContentType();
  if (f.length > 0) {
    writer.writeString(
      5,
      f
    );
  }
  f = message.getData_asU8();
  if (f.length > 0) {
    writer.writeBytes(
      7,
      f
    );
  }
  f = message.getTopic();
  if (f.length > 0) {
    writer.writeString(
      6,
      f
    );
  }
  f = message.getPubsubName();
  if (f.length > 0) {
    writer.writeString(
      8,
      f
    );
  }
  f = message.getPath();
  if (f.length > 0) {
    writer.writeString(
      9,
      f
    );
  }
};


/**
 * optional string id = 1;
 * @return {string}
 */
proto.dapr.proto.runtime.v1.TopicEventRequest.prototype.getId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.dapr.proto.runtime.v1.TopicEventRequest} returns this
 */
proto.dapr.proto.runtime.v1.TopicEventRequest.prototype.setId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string source = 2;
 * @return {string}
 */
proto.dapr.proto.runtime.v1.TopicEventRequest.prototype.getSource = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.dapr.proto.runtime.v1.TopicEventRequest} returns this
 */
proto.dapr.proto.runtime.v1.TopicEventRequest.prototype.setSource = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional string type = 3;
 * @return {string}
 */
proto.dapr.proto.runtime.v1.TopicEventRequest.prototype.getType = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.dapr.proto.runtime.v1.TopicEventRequest} returns this
 */
proto.dapr.proto.runtime.v1.TopicEventRequest.prototype.setType = function(value) {
  return jspb.Message.setProto3StringField(this, 3, value);
};


/**
 * optional string spec_version = 4;
 * @return {string}
 */
proto.dapr.proto.runtime.v1.TopicEventRequest.prototype.getSpecVersion = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 4, ""));
};


/**
 * @param {string} value
 * @return {!proto.dapr.proto.runtime.v1.TopicEventRequest} returns this
 */
proto.dapr.proto.runtime.v1.TopicEventRequest.prototype.setSpecVersion = function(value) {
  return jspb.Message.setProto3StringField(this, 4, value);
};


/**
 * optional string data_content_type = 5;
 * @return {string}
 */
proto.dapr.proto.runtime.v1.TopicEventRequest.prototype.getDataContentType = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 5, ""));
};


/**
 * @param {string} value
 * @return {!proto.dapr.proto.runtime.v1.TopicEventRequest} returns this
 */
proto.dapr.proto.runtime.v1.TopicEventRequest.prototype.setDataContentType = function(value) {
  return jspb.Message.setProto3StringField(this, 5, value);
};


/**
 * optional bytes data = 7;
 * @return {!(string|Uint8Array)}
 */
proto.dapr.proto.runtime.v1.TopicEventRequest.prototype.getData = function() {
  return /** @type {!(string|Uint8Array)} */ (jspb.Message.getFieldWithDefault(this, 7, ""));
};


/**
 * optional bytes data = 7;
 * This is a type-conversion wrapper around `getData()`
 * @return {string}
 */
proto.dapr.proto.runtime.v1.TopicEventRequest.prototype.getData_asB64 = function() {
  return /** @type {string} */ (jspb.Message.bytesAsB64(
      this.getData()));
};


/**
 * optional bytes data = 7;
 * Note that Uint8Array is not supported on all browsers.
 * @see http://caniuse.com/Uint8Array
 * This is a type-conversion wrapper around `getData()`
 * @return {!Uint8Array}
 */
proto.dapr.proto.runtime.v1.TopicEventRequest.prototype.getData_asU8 = function() {
  return /** @type {!Uint8Array} */ (jspb.Message.bytesAsU8(
      this.getData()));
};


/**
 * @param {!(string|Uint8Array)} value
 * @return {!proto.dapr.proto.runtime.v1.TopicEventRequest} returns this
 */
proto.dapr.proto.runtime.v1.TopicEventRequest.prototype.setData = function(value) {
  return jspb.Message.setProto3BytesField(this, 7, value);
};


/**
 * optional string topic = 6;
 * @return {string}
 */
proto.dapr.proto.runtime.v1.TopicEventRequest.prototype.getTopic = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 6, ""));
};


/**
 * @param {string} value
 * @return {!proto.dapr.proto.runtime.v1.TopicEventRequest} returns this
 */
proto.dapr.proto.runtime.v1.TopicEventRequest.prototype.setTopic = function(value) {
  return jspb.Message.setProto3StringField(this, 6, value);
};


/**
 * optional string pubsub_name = 8;
 * @return {string}
 */
proto.dapr.proto.runtime.v1.TopicEventRequest.prototype.getPubsubName = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 8, ""));
};


/**
 * @param {string} value
 * @return {!proto.dapr.proto.runtime.v1.TopicEventRequest} returns this
 */
proto.dapr.proto.runtime.v1.TopicEventRequest.prototype.setPubsubName = function(value) {
  return jspb.Message.setProto3StringField(this, 8, value);
};


/**
 * optional string path = 9;
 * @return {string}
 */
proto.dapr.proto.runtime.v1.TopicEventRequest.prototype.getPath = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 9, ""));
};


/**
 * @param {string} value
 * @return {!proto.dapr.proto.runtime.v1.TopicEventRequest} returns this
 */
proto.dapr.proto.runtime.v1.TopicEventRequest.prototype.setPath = function(value) {
  return jspb.Message.setProto3StringField(this, 9, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.dapr.proto.runtime.v1.TopicEventResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.dapr.proto.runtime.v1.TopicEventResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.dapr.proto.runtime.v1.TopicEventResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.dapr.proto.runtime.v1.TopicEventResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    status: jspb.Message.getFieldWithDefault(msg, 1, 0)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.dapr.proto.runtime.v1.TopicEventResponse}
 */
proto.dapr.proto.runtime.v1.TopicEventResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.dapr.proto.runtime.v1.TopicEventResponse;
  return proto.dapr.proto.runtime.v1.TopicEventResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.dapr.proto.runtime.v1.TopicEventResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.dapr.proto.runtime.v1.TopicEventResponse}
 */
proto.dapr.proto.runtime.v1.TopicEventResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {!proto.dapr.proto.runtime.v1.TopicEventResponse.TopicEventResponseStatus} */ (reader.readEnum());
      msg.setStatus(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.dapr.proto.runtime.v1.TopicEventResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.dapr.proto.runtime.v1.TopicEventResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.dapr.proto.runtime.v1.TopicEventResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.dapr.proto.runtime.v1.TopicEventResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getStatus();
  if (f !== 0.0) {
    writer.writeEnum(
      1,
      f
    );
  }
};


/**
 * @enum {number}
 */
proto.dapr.proto.runtime.v1.TopicEventResponse.TopicEventResponseStatus = {
  SUCCESS: 0,
  RETRY: 1,
  DROP: 2
};

/**
 * optional TopicEventResponseStatus status = 1;
 * @return {!proto.dapr.proto.runtime.v1.TopicEventResponse.TopicEventResponseStatus}
 */
proto.dapr.proto.runtime.v1.TopicEventResponse.prototype.getStatus = function() {
  return /** @type {!proto.dapr.proto.runtime.v1.TopicEventResponse.TopicEventResponseStatus} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {!proto.dapr.proto.runtime.v1.TopicEventResponse.TopicEventResponseStatus} value
 * @return {!proto.dapr.proto.runtime.v1.TopicEventResponse} returns this
 */
proto.dapr.proto.runtime.v1.TopicEventResponse.prototype.setStatus = function(value) {
  return jspb.Message.setProto3EnumField(this, 1, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.dapr.proto.runtime.v1.BindingEventRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.dapr.proto.runtime.v1.BindingEventRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.dapr.proto.runtime.v1.BindingEventRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.dapr.proto.runtime.v1.BindingEventRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    name: jspb.Message.getFieldWithDefault(msg, 1, ""),
    data: msg.getData_asB64(),
    metadataMap: (f = msg.getMetadataMap()) ? f.toObject(includeInstance, undefined) : []
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.dapr.proto.runtime.v1.BindingEventRequest}
 */
proto.dapr.proto.runtime.v1.BindingEventRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.dapr.proto.runtime.v1.BindingEventRequest;
  return proto.dapr.proto.runtime.v1.BindingEventRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.dapr.proto.runtime.v1.BindingEventRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.dapr.proto.runtime.v1.BindingEventRequest}
 */
proto.dapr.proto.runtime.v1.BindingEventRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setName(value);
      break;
    case 2:
      var value = /** @type {!Uint8Array} */ (reader.readBytes());
      msg.setData(value);
      break;
    case 3:
      var value = msg.getMetadataMap();
      reader.readMessage(value, function(message, reader) {
        jspb.Map.deserializeBinary(message, reader, jspb.BinaryReader.prototype.readString, jspb.BinaryReader.prototype.readString, null, "", "");
         });
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.dapr.proto.runtime.v1.BindingEventRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.dapr.proto.runtime.v1.BindingEventRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.dapr.proto.runtime.v1.BindingEventRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.dapr.proto.runtime.v1.BindingEventRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getName();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getData_asU8();
  if (f.length > 0) {
    writer.writeBytes(
      2,
      f
    );
  }
  f = message.getMetadataMap(true);
  if (f && f.getLength() > 0) {
    f.serializeBinary(3, writer, jspb.BinaryWriter.prototype.writeString, jspb.BinaryWriter.prototype.writeString);
  }
};


/**
 * optional string name = 1;
 * @return {string}
 */
proto.dapr.proto.runtime.v1.BindingEventRequest.prototype.getName = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.dapr.proto.runtime.v1.BindingEventRequest} returns this
 */
proto.dapr.proto.runtime.v1.BindingEventRequest.prototype.setName = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional bytes data = 2;
 * @return {!(string|Uint8Array)}
 */
proto.dapr.proto.runtime.v1.BindingEventRequest.prototype.getData = function() {
  return /** @type {!(string|Uint8Array)} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * optional bytes data = 2;
 * This is a type-conversion wrapper around `getData()`
 * @return {string}
 */
proto.dapr.proto.runtime.v1.BindingEventRequest.prototype.getData_asB64 = function() {
  return /** @type {string} */ (jspb.Message.bytesAsB64(
      this.getData()));
};


/**
 * optional bytes data = 2;
 * Note that Uint8Array is not supported on all browsers.
 * @see http://caniuse.com/Uint8Array
 * This is a type-conversion wrapper around `getData()`
 * @return {!Uint8Array}
 */
proto.dapr.proto.runtime.v1.BindingEventRequest.prototype.getData_asU8 = function() {
  return /** @type {!Uint8Array} */ (jspb.Message.bytesAsU8(
      this.getData()));
};


/**
 * @param {!(string|Uint8Array)} value
 * @return {!proto.dapr.proto.runtime.v1.BindingEventRequest} returns this
 */
proto.dapr.proto.runtime.v1.BindingEventRequest.prototype.setData = function(value) {
  return jspb.Message.setProto3BytesField(this, 2, value);
};


/**
 * map<string, string> metadata = 3;
 * @param {boolean=} opt_noLazyCreate Do not create the map if
 * empty, instead returning `undefined`
 * @return {!jspb.Map<string,string>}
 */
proto.dapr.proto.runtime.v1.BindingEventRequest.prototype.getMetadataMap = function(opt_noLazyCreate) {
  return /** @type {!jspb.Map<string,string>} */ (
      jspb.Message.getMapField(this, 3, opt_noLazyCreate,
      null));
};


/**
 * Clears values from the map. The map will be non-null.
 * @return {!proto.dapr.proto.runtime.v1.BindingEventRequest} returns this
 */
proto.dapr.proto.runtime.v1.BindingEventRequest.prototype.clearMetadataMap = function() {
  this.getMetadataMap().clear();
  return this;};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.dapr.proto.runtime.v1.BindingEventResponse.repeatedFields_ = [2,3];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.dapr.proto.runtime.v1.BindingEventResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.dapr.proto.runtime.v1.BindingEventResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.dapr.proto.runtime.v1.BindingEventResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.dapr.proto.runtime.v1.BindingEventResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    storeName: jspb.Message.getFieldWithDefault(msg, 1, ""),
    statesList: jspb.Message.toObjectList(msg.getStatesList(),
    dapr_proto_common_v1_common_pb.StateItem.toObject, includeInstance),
    toList: (f = jspb.Message.getRepeatedField(msg, 3)) == null ? undefined : f,
    data: msg.getData_asB64(),
    concurrency: jspb.Message.getFieldWithDefault(msg, 5, 0)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.dapr.proto.runtime.v1.BindingEventResponse}
 */
proto.dapr.proto.runtime.v1.BindingEventResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.dapr.proto.runtime.v1.BindingEventResponse;
  return proto.dapr.proto.runtime.v1.BindingEventResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.dapr.proto.runtime.v1.BindingEventResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.dapr.proto.runtime.v1.BindingEventResponse}
 */
proto.dapr.proto.runtime.v1.BindingEventResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setStoreName(value);
      break;
    case 2:
      var value = new dapr_proto_common_v1_common_pb.StateItem;
      reader.readMessage(value,dapr_proto_common_v1_common_pb.StateItem.deserializeBinaryFromReader);
      msg.addStates(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.addTo(value);
      break;
    case 4:
      var value = /** @type {!Uint8Array} */ (reader.readBytes());
      msg.setData(value);
      break;
    case 5:
      var value = /** @type {!proto.dapr.proto.runtime.v1.BindingEventResponse.BindingEventConcurrency} */ (reader.readEnum());
      msg.setConcurrency(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.dapr.proto.runtime.v1.BindingEventResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.dapr.proto.runtime.v1.BindingEventResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.dapr.proto.runtime.v1.BindingEventResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.dapr.proto.runtime.v1.BindingEventResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getStoreName();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getStatesList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      2,
      f,
      dapr_proto_common_v1_common_pb.StateItem.serializeBinaryToWriter
    );
  }
  f = message.getToList();
  if (f.length > 0) {
    writer.writeRepeatedString(
      3,
      f
    );
  }
  f = message.getData_asU8();
  if (f.length > 0) {
    writer.writeBytes(
      4,
      f
    );
  }
  f = message.getConcurrency();
  if (f !== 0.0) {
    writer.writeEnum(
      5,
      f
    );
  }
};


/**
 * @enum {number}
 */
proto.dapr.proto.runtime.v1.BindingEventResponse.BindingEventConcurrency = {
  SEQUENTIAL: 0,
  PARALLEL: 1
};

/**
 * optional string store_name = 1;
 * @return {string}
 */
proto.dapr.proto.runtime.v1.BindingEventResponse.prototype.getStoreName = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.dapr.proto.runtime.v1.BindingEventResponse} returns this
 */
proto.dapr.proto.runtime.v1.BindingEventResponse.prototype.setStoreName = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * repeated dapr.proto.common.v1.StateItem states = 2;
 * @return {!Array<!proto.dapr.proto.common.v1.StateItem>}
 */
proto.dapr.proto.runtime.v1.BindingEventResponse.prototype.getStatesList = function() {
  return /** @type{!Array<!proto.dapr.proto.common.v1.StateItem>} */ (
    jspb.Message.getRepeatedWrapperField(this, dapr_proto_common_v1_common_pb.StateItem, 2));
};


/**
 * @param {!Array<!proto.dapr.proto.common.v1.StateItem>} value
 * @return {!proto.dapr.proto.runtime.v1.BindingEventResponse} returns this
*/
proto.dapr.proto.runtime.v1.BindingEventResponse.prototype.setStatesList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 2, value);
};


/**
 * @param {!proto.dapr.proto.common.v1.StateItem=} opt_value
 * @param {number=} opt_index
 * @return {!proto.dapr.proto.common.v1.StateItem}
 */
proto.dapr.proto.runtime.v1.BindingEventResponse.prototype.addStates = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 2, opt_value, proto.dapr.proto.common.v1.StateItem, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.dapr.proto.runtime.v1.BindingEventResponse} returns this
 */
proto.dapr.proto.runtime.v1.BindingEventResponse.prototype.clearStatesList = function() {
  return this.setStatesList([]);
};


/**
 * repeated string to = 3;
 * @return {!Array<string>}
 */
proto.dapr.proto.runtime.v1.BindingEventResponse.prototype.getToList = function() {
  return /** @type {!Array<string>} */ (jspb.Message.getRepeatedField(this, 3));
};


/**
 * @param {!Array<string>} value
 * @return {!proto.dapr.proto.runtime.v1.BindingEventResponse} returns this
 */
proto.dapr.proto.runtime.v1.BindingEventResponse.prototype.setToList = function(value) {
  return jspb.Message.setField(this, 3, value || []);
};


/**
 * @param {string} value
 * @param {number=} opt_index
 * @return {!proto.dapr.proto.runtime.v1.BindingEventResponse} returns this
 */
proto.dapr.proto.runtime.v1.BindingEventResponse.prototype.addTo = function(value, opt_index) {
  return jspb.Message.addToRepeatedField(this, 3, value, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.dapr.proto.runtime.v1.BindingEventResponse} returns this
 */
proto.dapr.proto.runtime.v1.BindingEventResponse.prototype.clearToList = function() {
  return this.setToList([]);
};


/**
 * optional bytes data = 4;
 * @return {!(string|Uint8Array)}
 */
proto.dapr.proto.runtime.v1.BindingEventResponse.prototype.getData = function() {
  return /** @type {!(string|Uint8Array)} */ (jspb.Message.getFieldWithDefault(this, 4, ""));
};


/**
 * optional bytes data = 4;
 * This is a type-conversion wrapper around `getData()`
 * @return {string}
 */
proto.dapr.proto.runtime.v1.BindingEventResponse.prototype.getData_asB64 = function() {
  return /** @type {string} */ (jspb.Message.bytesAsB64(
      this.getData()));
};


/**
 * optional bytes data = 4;
 * Note that Uint8Array is not supported on all browsers.
 * @see http://caniuse.com/Uint8Array
 * This is a type-conversion wrapper around `getData()`
 * @return {!Uint8Array}
 */
proto.dapr.proto.runtime.v1.BindingEventResponse.prototype.getData_asU8 = function() {
  return /** @type {!Uint8Array} */ (jspb.Message.bytesAsU8(
      this.getData()));
};


/**
 * @param {!(string|Uint8Array)} value
 * @return {!proto.dapr.proto.runtime.v1.BindingEventResponse} returns this
 */
proto.dapr.proto.runtime.v1.BindingEventResponse.prototype.setData = function(value) {
  return jspb.Message.setProto3BytesField(this, 4, value);
};


/**
 * optional BindingEventConcurrency concurrency = 5;
 * @return {!proto.dapr.proto.runtime.v1.BindingEventResponse.BindingEventConcurrency}
 */
proto.dapr.proto.runtime.v1.BindingEventResponse.prototype.getConcurrency = function() {
  return /** @type {!proto.dapr.proto.runtime.v1.BindingEventResponse.BindingEventConcurrency} */ (jspb.Message.getFieldWithDefault(this, 5, 0));
};


/**
 * @param {!proto.dapr.proto.runtime.v1.BindingEventResponse.BindingEventConcurrency} value
 * @return {!proto.dapr.proto.runtime.v1.BindingEventResponse} returns this
 */
proto.dapr.proto.runtime.v1.BindingEventResponse.prototype.setConcurrency = function(value) {
  return jspb.Message.setProto3EnumField(this, 5, value);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.dapr.proto.runtime.v1.ListTopicSubscriptionsResponse.repeatedFields_ = [1];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.dapr.proto.runtime.v1.ListTopicSubscriptionsResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.dapr.proto.runtime.v1.ListTopicSubscriptionsResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.dapr.proto.runtime.v1.ListTopicSubscriptionsResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.dapr.proto.runtime.v1.ListTopicSubscriptionsResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    subscriptionsList: jspb.Message.toObjectList(msg.getSubscriptionsList(),
    proto.dapr.proto.runtime.v1.TopicSubscription.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.dapr.proto.runtime.v1.ListTopicSubscriptionsResponse}
 */
proto.dapr.proto.runtime.v1.ListTopicSubscriptionsResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.dapr.proto.runtime.v1.ListTopicSubscriptionsResponse;
  return proto.dapr.proto.runtime.v1.ListTopicSubscriptionsResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.dapr.proto.runtime.v1.ListTopicSubscriptionsResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.dapr.proto.runtime.v1.ListTopicSubscriptionsResponse}
 */
proto.dapr.proto.runtime.v1.ListTopicSubscriptionsResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new proto.dapr.proto.runtime.v1.TopicSubscription;
      reader.readMessage(value,proto.dapr.proto.runtime.v1.TopicSubscription.deserializeBinaryFromReader);
      msg.addSubscriptions(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.dapr.proto.runtime.v1.ListTopicSubscriptionsResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.dapr.proto.runtime.v1.ListTopicSubscriptionsResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.dapr.proto.runtime.v1.ListTopicSubscriptionsResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.dapr.proto.runtime.v1.ListTopicSubscriptionsResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getSubscriptionsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      1,
      f,
      proto.dapr.proto.runtime.v1.TopicSubscription.serializeBinaryToWriter
    );
  }
};


/**
 * repeated TopicSubscription subscriptions = 1;
 * @return {!Array<!proto.dapr.proto.runtime.v1.TopicSubscription>}
 */
proto.dapr.proto.runtime.v1.ListTopicSubscriptionsResponse.prototype.getSubscriptionsList = function() {
  return /** @type{!Array<!proto.dapr.proto.runtime.v1.TopicSubscription>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.dapr.proto.runtime.v1.TopicSubscription, 1));
};


/**
 * @param {!Array<!proto.dapr.proto.runtime.v1.TopicSubscription>} value
 * @return {!proto.dapr.proto.runtime.v1.ListTopicSubscriptionsResponse} returns this
*/
proto.dapr.proto.runtime.v1.ListTopicSubscriptionsResponse.prototype.setSubscriptionsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 1, value);
};


/**
 * @param {!proto.dapr.proto.runtime.v1.TopicSubscription=} opt_value
 * @param {number=} opt_index
 * @return {!proto.dapr.proto.runtime.v1.TopicSubscription}
 */
proto.dapr.proto.runtime.v1.ListTopicSubscriptionsResponse.prototype.addSubscriptions = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 1, opt_value, proto.dapr.proto.runtime.v1.TopicSubscription, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.dapr.proto.runtime.v1.ListTopicSubscriptionsResponse} returns this
 */
proto.dapr.proto.runtime.v1.ListTopicSubscriptionsResponse.prototype.clearSubscriptionsList = function() {
  return this.setSubscriptionsList([]);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.dapr.proto.runtime.v1.TopicSubscription.prototype.toObject = function(opt_includeInstance) {
  return proto.dapr.proto.runtime.v1.TopicSubscription.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.dapr.proto.runtime.v1.TopicSubscription} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.dapr.proto.runtime.v1.TopicSubscription.toObject = function(includeInstance, msg) {
  var f, obj = {
    pubsubName: jspb.Message.getFieldWithDefault(msg, 1, ""),
    topic: jspb.Message.getFieldWithDefault(msg, 2, ""),
    metadataMap: (f = msg.getMetadataMap()) ? f.toObject(includeInstance, undefined) : [],
    routes: (f = msg.getRoutes()) && proto.dapr.proto.runtime.v1.TopicRoutes.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.dapr.proto.runtime.v1.TopicSubscription}
 */
proto.dapr.proto.runtime.v1.TopicSubscription.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.dapr.proto.runtime.v1.TopicSubscription;
  return proto.dapr.proto.runtime.v1.TopicSubscription.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.dapr.proto.runtime.v1.TopicSubscription} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.dapr.proto.runtime.v1.TopicSubscription}
 */
proto.dapr.proto.runtime.v1.TopicSubscription.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setPubsubName(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setTopic(value);
      break;
    case 3:
      var value = msg.getMetadataMap();
      reader.readMessage(value, function(message, reader) {
        jspb.Map.deserializeBinary(message, reader, jspb.BinaryReader.prototype.readString, jspb.BinaryReader.prototype.readString, null, "", "");
         });
      break;
    case 5:
      var value = new proto.dapr.proto.runtime.v1.TopicRoutes;
      reader.readMessage(value,proto.dapr.proto.runtime.v1.TopicRoutes.deserializeBinaryFromReader);
      msg.setRoutes(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.dapr.proto.runtime.v1.TopicSubscription.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.dapr.proto.runtime.v1.TopicSubscription.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.dapr.proto.runtime.v1.TopicSubscription} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.dapr.proto.runtime.v1.TopicSubscription.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getPubsubName();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getTopic();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getMetadataMap(true);
  if (f && f.getLength() > 0) {
    f.serializeBinary(3, writer, jspb.BinaryWriter.prototype.writeString, jspb.BinaryWriter.prototype.writeString);
  }
  f = message.getRoutes();
  if (f != null) {
    writer.writeMessage(
      5,
      f,
      proto.dapr.proto.runtime.v1.TopicRoutes.serializeBinaryToWriter
    );
  }
};


/**
 * optional string pubsub_name = 1;
 * @return {string}
 */
proto.dapr.proto.runtime.v1.TopicSubscription.prototype.getPubsubName = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.dapr.proto.runtime.v1.TopicSubscription} returns this
 */
proto.dapr.proto.runtime.v1.TopicSubscription.prototype.setPubsubName = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string topic = 2;
 * @return {string}
 */
proto.dapr.proto.runtime.v1.TopicSubscription.prototype.getTopic = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.dapr.proto.runtime.v1.TopicSubscription} returns this
 */
proto.dapr.proto.runtime.v1.TopicSubscription.prototype.setTopic = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * map<string, string> metadata = 3;
 * @param {boolean=} opt_noLazyCreate Do not create the map if
 * empty, instead returning `undefined`
 * @return {!jspb.Map<string,string>}
 */
proto.dapr.proto.runtime.v1.TopicSubscription.prototype.getMetadataMap = function(opt_noLazyCreate) {
  return /** @type {!jspb.Map<string,string>} */ (
      jspb.Message.getMapField(this, 3, opt_noLazyCreate,
      null));
};


/**
 * Clears values from the map. The map will be non-null.
 * @return {!proto.dapr.proto.runtime.v1.TopicSubscription} returns this
 */
proto.dapr.proto.runtime.v1.TopicSubscription.prototype.clearMetadataMap = function() {
  this.getMetadataMap().clear();
  return this;};


/**
 * optional TopicRoutes routes = 5;
 * @return {?proto.dapr.proto.runtime.v1.TopicRoutes}
 */
proto.dapr.proto.runtime.v1.TopicSubscription.prototype.getRoutes = function() {
  return /** @type{?proto.dapr.proto.runtime.v1.TopicRoutes} */ (
    jspb.Message.getWrapperField(this, proto.dapr.proto.runtime.v1.TopicRoutes, 5));
};


/**
 * @param {?proto.dapr.proto.runtime.v1.TopicRoutes|undefined} value
 * @return {!proto.dapr.proto.runtime.v1.TopicSubscription} returns this
*/
proto.dapr.proto.runtime.v1.TopicSubscription.prototype.setRoutes = function(value) {
  return jspb.Message.setWrapperField(this, 5, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.dapr.proto.runtime.v1.TopicSubscription} returns this
 */
proto.dapr.proto.runtime.v1.TopicSubscription.prototype.clearRoutes = function() {
  return this.setRoutes(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.dapr.proto.runtime.v1.TopicSubscription.prototype.hasRoutes = function() {
  return jspb.Message.getField(this, 5) != null;
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.dapr.proto.runtime.v1.TopicRoutes.repeatedFields_ = [1];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.dapr.proto.runtime.v1.TopicRoutes.prototype.toObject = function(opt_includeInstance) {
  return proto.dapr.proto.runtime.v1.TopicRoutes.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.dapr.proto.runtime.v1.TopicRoutes} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.dapr.proto.runtime.v1.TopicRoutes.toObject = function(includeInstance, msg) {
  var f, obj = {
    rulesList: jspb.Message.toObjectList(msg.getRulesList(),
    proto.dapr.proto.runtime.v1.TopicRule.toObject, includeInstance),
    pb_default: jspb.Message.getFieldWithDefault(msg, 2, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.dapr.proto.runtime.v1.TopicRoutes}
 */
proto.dapr.proto.runtime.v1.TopicRoutes.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.dapr.proto.runtime.v1.TopicRoutes;
  return proto.dapr.proto.runtime.v1.TopicRoutes.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.dapr.proto.runtime.v1.TopicRoutes} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.dapr.proto.runtime.v1.TopicRoutes}
 */
proto.dapr.proto.runtime.v1.TopicRoutes.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new proto.dapr.proto.runtime.v1.TopicRule;
      reader.readMessage(value,proto.dapr.proto.runtime.v1.TopicRule.deserializeBinaryFromReader);
      msg.addRules(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setDefault(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.dapr.proto.runtime.v1.TopicRoutes.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.dapr.proto.runtime.v1.TopicRoutes.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.dapr.proto.runtime.v1.TopicRoutes} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.dapr.proto.runtime.v1.TopicRoutes.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getRulesList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      1,
      f,
      proto.dapr.proto.runtime.v1.TopicRule.serializeBinaryToWriter
    );
  }
  f = message.getDefault();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
};


/**
 * repeated TopicRule rules = 1;
 * @return {!Array<!proto.dapr.proto.runtime.v1.TopicRule>}
 */
proto.dapr.proto.runtime.v1.TopicRoutes.prototype.getRulesList = function() {
  return /** @type{!Array<!proto.dapr.proto.runtime.v1.TopicRule>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.dapr.proto.runtime.v1.TopicRule, 1));
};


/**
 * @param {!Array<!proto.dapr.proto.runtime.v1.TopicRule>} value
 * @return {!proto.dapr.proto.runtime.v1.TopicRoutes} returns this
*/
proto.dapr.proto.runtime.v1.TopicRoutes.prototype.setRulesList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 1, value);
};


/**
 * @param {!proto.dapr.proto.runtime.v1.TopicRule=} opt_value
 * @param {number=} opt_index
 * @return {!proto.dapr.proto.runtime.v1.TopicRule}
 */
proto.dapr.proto.runtime.v1.TopicRoutes.prototype.addRules = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 1, opt_value, proto.dapr.proto.runtime.v1.TopicRule, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.dapr.proto.runtime.v1.TopicRoutes} returns this
 */
proto.dapr.proto.runtime.v1.TopicRoutes.prototype.clearRulesList = function() {
  return this.setRulesList([]);
};


/**
 * optional string default = 2;
 * @return {string}
 */
proto.dapr.proto.runtime.v1.TopicRoutes.prototype.getDefault = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.dapr.proto.runtime.v1.TopicRoutes} returns this
 */
proto.dapr.proto.runtime.v1.TopicRoutes.prototype.setDefault = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.dapr.proto.runtime.v1.TopicRule.prototype.toObject = function(opt_includeInstance) {
  return proto.dapr.proto.runtime.v1.TopicRule.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.dapr.proto.runtime.v1.TopicRule} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.dapr.proto.runtime.v1.TopicRule.toObject = function(includeInstance, msg) {
  var f, obj = {
    match: jspb.Message.getFieldWithDefault(msg, 1, ""),
    path: jspb.Message.getFieldWithDefault(msg, 2, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.dapr.proto.runtime.v1.TopicRule}
 */
proto.dapr.proto.runtime.v1.TopicRule.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.dapr.proto.runtime.v1.TopicRule;
  return proto.dapr.proto.runtime.v1.TopicRule.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.dapr.proto.runtime.v1.TopicRule} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.dapr.proto.runtime.v1.TopicRule}
 */
proto.dapr.proto.runtime.v1.TopicRule.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setMatch(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setPath(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.dapr.proto.runtime.v1.TopicRule.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.dapr.proto.runtime.v1.TopicRule.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.dapr.proto.runtime.v1.TopicRule} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.dapr.proto.runtime.v1.TopicRule.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getMatch();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getPath();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
};


/**
 * optional string match = 1;
 * @return {string}
 */
proto.dapr.proto.runtime.v1.TopicRule.prototype.getMatch = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.dapr.proto.runtime.v1.TopicRule} returns this
 */
proto.dapr.proto.runtime.v1.TopicRule.prototype.setMatch = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string path = 2;
 * @return {string}
 */
proto.dapr.proto.runtime.v1.TopicRule.prototype.getPath = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.dapr.proto.runtime.v1.TopicRule} returns this
 */
proto.dapr.proto.runtime.v1.TopicRule.prototype.setPath = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.dapr.proto.runtime.v1.ListInputBindingsResponse.repeatedFields_ = [1];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.dapr.proto.runtime.v1.ListInputBindingsResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.dapr.proto.runtime.v1.ListInputBindingsResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.dapr.proto.runtime.v1.ListInputBindingsResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.dapr.proto.runtime.v1.ListInputBindingsResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    bindingsList: (f = jspb.Message.getRepeatedField(msg, 1)) == null ? undefined : f
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.dapr.proto.runtime.v1.ListInputBindingsResponse}
 */
proto.dapr.proto.runtime.v1.ListInputBindingsResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.dapr.proto.runtime.v1.ListInputBindingsResponse;
  return proto.dapr.proto.runtime.v1.ListInputBindingsResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.dapr.proto.runtime.v1.ListInputBindingsResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.dapr.proto.runtime.v1.ListInputBindingsResponse}
 */
proto.dapr.proto.runtime.v1.ListInputBindingsResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.addBindings(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.dapr.proto.runtime.v1.ListInputBindingsResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.dapr.proto.runtime.v1.ListInputBindingsResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.dapr.proto.runtime.v1.ListInputBindingsResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.dapr.proto.runtime.v1.ListInputBindingsResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getBindingsList();
  if (f.length > 0) {
    writer.writeRepeatedString(
      1,
      f
    );
  }
};


/**
 * repeated string bindings = 1;
 * @return {!Array<string>}
 */
proto.dapr.proto.runtime.v1.ListInputBindingsResponse.prototype.getBindingsList = function() {
  return /** @type {!Array<string>} */ (jspb.Message.getRepeatedField(this, 1));
};


/**
 * @param {!Array<string>} value
 * @return {!proto.dapr.proto.runtime.v1.ListInputBindingsResponse} returns this
 */
proto.dapr.proto.runtime.v1.ListInputBindingsResponse.prototype.setBindingsList = function(value) {
  return jspb.Message.setField(this, 1, value || []);
};


/**
 * @param {string} value
 * @param {number=} opt_index
 * @return {!proto.dapr.proto.runtime.v1.ListInputBindingsResponse} returns this
 */
proto.dapr.proto.runtime.v1.ListInputBindingsResponse.prototype.addBindings = function(value, opt_index) {
  return jspb.Message.addToRepeatedField(this, 1, value, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.dapr.proto.runtime.v1.ListInputBindingsResponse} returns this
 */
proto.dapr.proto.runtime.v1.ListInputBindingsResponse.prototype.clearBindingsList = function() {
  return this.setBindingsList([]);
};


goog.object.extend(exports, proto.dapr.proto.runtime.v1);
